### PR TITLE
qa: Remove race between connecting and shutdown on separate connections

### PR DESCRIPTION
**Pull request by promag**
Fixes the error https://github.com/bitcoin/bitcoin/pull/14670#issuecomment-447255352 reported by @ken2812221.

There is a race between RPC stop and another concurrent call in the test framework. The connection must be established and the command `waitfornewblock` running before calling `stop`.

See also https://github.com/bitcoin/bitcoin/pull/14670#issuecomment-447304513.

### CODE REVIEW

File: test/functional/feature_shutdown.py [hunk 1]
```diff
@@ -20,7 +22,10 @@ def set_test_params(self):
 
     def run_test(self):
         node = get_rpc_proxy(self.nodes[0].url, 1, timeout=600, coveragedir=self.nodes[0].coverage_dir)
+        # force connection establishment
+        node.getblockcount()
         Thread(target=test_long_call, args=(node,)).start()
+        sleep(1)
```
Comment (laanwj) [inline test/functional/feature_shutdown.py]: I don't like hardcoding a second of sleep here, hardly seems robust.

Comment (promag) [reply test/functional/feature_shutdown.py]: Agree, this should be replaced by something like:
```py
wait_until(lambda: self.getrpcinfo()['active_workers'] > 1)
```
Other suggestions?

File: src/rpc/server.cpp [hunk 2]
```diff
@@ -31,12 +31,12 @@ static std::string rpcWarmupStatus GUARDED_BY(cs_rpcWarmup) = "RPC server starte
 static RPCTimerInterface* timerInterface = nullptr;
 /* Map of name to timer. */
 static std::map<std::string, std::unique_ptr<RPCTimerBase> > deadlineTimers;
+static std::shared_ptr<int> g_active_commands_counter = std::make_shared<int>(0);
```
Comment (promag) [inline src/rpc/server.cpp]: Consider this approach temporary, just for concept ack.

File: test/functional/feature_shutdown.py [hunk 3]
```diff
@@ -20,8 +20,14 @@ def set_test_params(self):
 
     def run_test(self):
         node = get_rpc_proxy(self.nodes[0].url, 1, timeout=600, coveragedir=self.nodes[0].coverage_dir)
+        # Force connection establishment by executing a dummy command.
+        node.getblockcount()
```
Comment (MarcoFalke) [inline test/functional/feature_shutdown.py]: Is this required given that we wait anyway until node.waitfornewblock is called, which in turn should set up the connection?

Comment (promag) [reply test/functional/feature_shutdown.py]: For the test to pass this is not required, but this is here to note that there's still a race after connection establishment. Let me know if it's not important.

**Comment by MarcoFalke** [comment]
utACK 49a79d454284df7758f83ccd3121afb3e6799342

**Comment by laanwj** [comment]
utACK 4412a59bfe8228698e5b5bbe8bb21c8e8a70d357
No difference with 49a79d4 except for commit msg.

**Pull request merged**